engine/src/system.h

```diff
@@ -188,7 +188,7 @@ class MCMemoryFileHandle: public MCSystemFileHandle
 		// If there isn't enough room, extend
 		if (m_pointer + p_length > m_capacity)
```

Comment:
Ooh, I just noticed that this appears to be missing an overflow check when calculating the new capacity.